### PR TITLE
add hex integer literal parsing support

### PR DESCRIPTION
**Pull request by kaspar030**
This PR implements parsing integer literals as both decimal and hex (`1234` and `0x1234`).
~~It adds `num_traits` as dependency, but that's a very light one.~~

### CODE REVIEW

File: src/token/mod.rs [hunk 1]
```diff
@@ -348,7 +348,7 @@ fn partial_tokens_to_tokens(mut tokens: &[PartialToken]) -> EvalexprResult<Vec<T
                 },
                 PartialToken::Literal(literal) => {
                     cutoff = 1;
-                    if let Ok(number) = literal.parse::<IntType>() {
+                    if let Ok(number) = literal.parse_dec_or_hex::<IntType>() {
```
Comment (ISibboI) [inline src/token/mod.rs:351]: Can we do this without `num-traits` by using `IntType::from_str_radix` instead?

By first checking if the prefix is "0x", and if yes, parsing as a hex, and if no, then continuing the if/else chain.

File: tests/integration.rs [hunk 2]
```diff
@@ -2291,3 +2291,9 @@ fn test_builtin_functions_context() {
         )))
     );
 }
+
+#[test]
+fn test_hex() {
```
Comment (ISibboI) [inline tests/integration.rs:2296]: Should also test negative numbers in this test, as well as the string "0x" (which should return an error).

**Review by ISibboI — COMMENTED** [review]
Great idea, thanks! Can we do this without `num-traits` as described in the comment?

File: tests/integration.rs (hunk 2 above)
Comment (kaspar030) [reply tests/integration.rs:2296]: Which error would be appropriate for the "0x" case? Currently it falls back to `EvalexprError::VariableIdentifierNotFound("0x")`.

File: src/token/mod.rs (hunk 1 above)
Comment (kaspar030) [reply src/token/mod.rs]: yeah, probably better. I somehow tried to stay generic over the used IntType.

**Comment by ISibboI** [comment]
Could you fix the two lints? Then we can merge this.

**Comment by ISibboI** [comment]
Great, thanks!
